### PR TITLE
📞 Cleanup ChannelConnection and add index for IVR retries

### PR DESCRIPTION
**Pull request by rowanseymour**
Goes with https://github.com/nyaruka/mailroom/pull/477

### CODE REVIEW

File: temba/channels/migrations/0129_auto_20210810_2041.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Generated by Django 3.2.6 on 2021-08-10 20:41
```
Comment (rowanseymour) [inline temba/channels/migrations/0129_auto_20210810_2041.py:1]:
```
BEGIN;
--
-- Alter field created_on on channelconnection
--
--
-- Alter field duration on channelconnection
--
--
-- Alter field ended_on on channelconnection
--
--
-- Alter field error_count on channelconnection
--
--
-- Alter field external_id on channelconnection
--
--
-- Alter field modified_on on channelconnection
--
--
-- Alter field next_attempt on channelconnection
--
--
-- Alter field retry_count on channelconnection
--
--
-- Alter field started_on on channelconnection
--
--
-- Alter field status on channelconnection
--
--
-- Create index channelconnection_ivr_to_retry on field(s) id, next_attempt of model channelconnection
--
CREATE INDEX "channelconnection_ivr_to_retry" 
ON "channels_channelconnection" ("id", "next_attempt") 
WHERE ("next_attempt" IS NOT NULL AND "status" IN ('N', 'B', 'E'));
COMMIT
```

File: temba/channels/migrations/0129_auto_20210810_2041.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                    ("F", "Failed"),
+                    ("E", "Errored"),
+                ],
+                max_length=1,
+            ),
+        ),
+        migrations.AddIndex(
+            model_name="channelconnection",
+            index=models.Index(
+                condition=models.Q(("next_attempt__isnull", False), ("status__in", ("N", "B", "E"))),
```
Comment (rowanseymour) [inline temba/channels/migrations/0129_auto_20210810_2041.py:80]: currently mailroom looks for connections with E and Q statuses.. but we don't use Q for that and I'd like to use N for modelling voicemail

File: temba/channels/models.py [hunk 3]
```diff
@@ -1680,84 +1680,53 @@ class ChannelConnection(models.Model):
...
-    FAILED = "F"  # the platform failed to initiate the call (bad phone number)
-    NO_ANSWER = "N"  # the call has timed-out or ringed-out
-    CANCELED = "C"  # the call was terminated by platform
-    COMPLETED = "D"  # the call was completed successfully
-
-    # valid for USSD sessions
-    TRIGGERED = "T"
-    INTERRUPTED = "X"
-    INITIATED = "A"
-    ENDING = "E"
```
Comment (rowanseymour) [inline temba/channels/models.py:1700]: not what we currently use E for...

File: temba/channels/models.py [hunk 4]
```diff
@@ -1680,84 +1680,53 @@ class ChannelConnection(models.Model):
     Base for IVR sessions which require a connection to specific channel
     """
 
-    PENDING = "P"  # initial state for all sessions
-    QUEUED = "Q"  # the session is queued internally
```
Comment (rowanseymour) [inline temba/channels/models.py:1684]: P and Q aren't used as far as I can see

File: temba/sql/current_indexes.sql [hunk 5]
```diff
@@ -17,9 +17,6 @@ ON channels_channelcount(channel_id, count_type, day) WHERE NOT is_squashed;
 CREATE INDEX channels_channellog_channel_created_on
 ON channels_channellog(channel_id, created_on desc);
 
-CREATE INDEX IF NOT EXISTS channels_channelsession_pending_events
-  ON channels_channelconnection (modified_on ASC NULLS LAST) WHERE status = 'P' and direction = 'O';
```
Comment (rowanseymour) [inline temba/sql/current_indexes.sql:21]: assuming this had a use for USSD

File: temba/channels/models.py [hunk 6]
```diff
@@ -1680,84 +1680,56 @@ class ChannelConnection(models.Model):
...
+    created_on = models.DateTimeField(default=timezone.now)
+    modified_on = models.DateTimeField(default=timezone.now)
+    started_on = models.DateTimeField(null=True)
+    ended_on = models.DateTimeField(null=True)
+    duration = models.IntegerField(null=True)  # in seconds
+
+    retry_count = models.IntegerField()
+    next_attempt = models.DateTimeField(null=True)
+
+    error_count = models.IntegerField(null=True)  # TODO remove, mailroom uses retry_count
```
Comment (rowanseymour) [inline temba/channels/models.py:1732]: mailroom uses `retry_count` which seems nicer in the cases when we want to retry something that isn't actually an error.. e.g. no answer

**Review by norkans7 — APPROVED** [review]

**Pull request merged**